File: Code/Factory.Java/src/factory/basic/ConcreteFactory2.java
package factory.basic;

import factory.entities.ConcreteThing2;
import factory.interfaces.IFactory;
import factory.interfaces.IThing;
/**
 * Factory that creates objects of the ConcreteThing2 class
 */
public class ConcreteFactory2 implements IFactory {

	@Override
	public IThing createThing() {
		return new ConcreteThing2();
	}

}

File: Code/Factory.dotNet/Factory.dotNet/Basic/ConcreteFactory2.cs
using Factory.Interfaces;
using Factory.Models;

namespace Factory.Basic
{
    /// <summary>
    /// Factory that creates objects of the ConcreteThing2 class
    /// </summary>
    public class ConcreteFactory2 : IFactory
    {
        public IThing CreateThing()
        {
            return new ConcreteThing2();
        }
    }
}
